axelor-sale/src/main/java/com/axelor/apps/sale/service/ProductRestServiceImpl.java

```diff
@@ -0,0 +1,195 @@
+    }
+    if (accountManagement.getSaleTaxSet() == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE,
+          String.format(I18n.get(SaleExceptionMessage.NO_PRODUCT_TAX), product.getId().toString()));
+    }
+
+    Tax tax = accountManagement.getSaleTaxSet().stream().findFirst().get();
+    if (tax.getActiveTaxLine() == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE,
+          String.format(SaleExceptionMessage.NO_TAX_LINE, tax.getId().toString()));
+    }
+    BigDecimal taxValue = tax.getActiveTaxLine().getValue();
+    BigDecimal priceConverted = priceInPartnerCurrency(product, partner, company);
+    return priceConverted.add(priceConverted.multiply(taxValue.divide(BigDecimal.valueOf(100))));
+  }
+
+  protected void checkProduct(Product product) throws AxelorException {
+    if (product == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE, I18n.get(SaleExceptionMessage.PRODUCT_IS_NULL));
+    }
+  }
+
+  @Override
+  public CurrencyResponse createCurrencyResponse(Product product, Partner partner, Company company)
+      throws AxelorException {
+    if (partner != null && partner.getCurrency() != null)
+      return new CurrencyResponse(partner.getCurrency());
+    if (company != null && company.getCurrency() != null)
+      return new CurrencyResponse(company.getCurrency());
+    if (product.getSaleCurrency() == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE,
+          I18n.get(SaleExceptionMessage.PRODUCT_CURRENCY_IS_NULL));
+    }
+    return new CurrencyResponse(product.getSaleCurrency());
+  }
+
+  @Override
+  public ProductResponse computeProductResponse(Company company, Product product, Partner partner)
+      throws AxelorException, JSONException {
+    CurrencyResponse currencyResponse = createCurrencyResponse(product, partner, company);
+    List<PriceResponse> prices = fetchProductPrice(product, partner, company);
+    return new ProductResponse(product.getId(), prices, currencyResponse);
+  }
+
+  protected BigDecimal priceInPartnerCurrency(Product product, Partner partner, Company company)
+      throws AxelorException {
+    if (product == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE, I18n.get(SaleExceptionMessage.PRODUCT_IS_NULL));
+    }
+
+    BigDecimal price = product.getSalePrice();
+
+    Currency partnerCurrency = partner.getCurrency();
+
+    Currency productCurrency = product.getSaleCurrency();
+
+    Currency companyCurrency = company.getCurrency();
+
+    LocalDate localDate =
+        appBaseService.getTodayDate(
+            Optional.ofNullable(AuthUtils.getUser()).map(User::getActiveCompany).orElse(null));
+    if (partnerCurrency == null)
+      return Beans.get(CurrencyService.class)
+          .getAmountCurrencyConvertedAtDate(productCurrency, companyCurrency, price, localDate);
```

Comment:
In AOS, we always use brackets for a if-block, even if there is only one line. Please add the brackets.
Also remove Beans.get.